### PR TITLE
[doc] Update acknowledgement and dead letter topic documentation.

### LINKED ISSUE #4119: Documentation for Dead letter topic
This is a task to track documentation of dead letter topic.

### PR DESCRIPTION
**Pull request by codelipenghui**
Fixes #4119

### Motivation

Catch up documentation of dead letter topic.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (yes)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by codelipenghui** [comment]
@Jennifer88huang Please help to review the documentation, thanks.

**Comment by sijie** [comment]
@Jennifer88huang @Anonymitaet

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -72,9 +76,42 @@ Messages can be acknowledged either one by one or cumulatively. With cumulative
...
 
-### Listeners
+In Shared subscription mode, messages can be acknowledged individually.
 
-Client libraries can provide their own listener implementations for consumers. The [Java client](client-libraries-java.md), for example, provides a {@inject: javadoc:MesssageListener:/client/org/apache/pulsar/client/api/MessageListener} interface. In this interface, the `received` method is called whenever a new message is received.
+### Negative Acknowledgement
+
+When a consumer can't successfully process a message at this time and looking forward to another chance to process the message, consumer can send a negative acknowledgement to the broker so that the broker can redeliver the message.
+
+Messages can be negative acknowledged one by one or cumulatively or negative acknowledged individually, this is related to the consumption subscription mode.
```
[the remark this answers is not in the document]
Comment (codelipenghui) [reply site2/docs/concepts-messaging.md]: one by one has sequential semantics.

File: site2/docs/concepts-messaging.md [hunk 2]
```diff
@@ -72,9 +76,42 @@ Messages can be acknowledged either one by one or cumulatively. With cumulative
 
 > Cumulative acknowledgement cannot be used with [shared subscription mode](#subscription-modes), because shared mode involves multiple consumers having access to the same subscription.
 
-### Listeners
+In Shared subscription mode, messages can be acknowledged individually.
 
-Client libraries can provide their own listener implementations for consumers. The [Java client](client-libraries-java.md), for example, provides a {@inject: javadoc:MesssageListener:/client/org/apache/pulsar/client/api/MessageListener} interface. In this interface, the `received` method is called whenever a new message is received.
+### Negative Acknowledgement
+
+When a consumer can't successfully process a message at this time and looking forward to another chance to process the message, consumer can send a negative acknowledgement to the broker so that the broker can redeliver the message.
```
[the remark this answers is not in the document]
Comment (codelipenghui) [reply site2/docs/concepts-messaging.md]: yes

[the remark this answers is not in the document]
Comment (Jennifer88huang) [reply site2/docs/concepts-messaging.md]: Thank you for your quick response, if so, we can use "consume a message" here.

File: site2/docs/concepts-messaging.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jennifer88huang) [reply site2/docs/concepts-messaging.md]: so could we refine it as  "Messages can be negatively acknowledged one by one or cumulatively, which depends on the consumption subscription mode."

**Comment by codelipenghui** [comment]
@Jennifer88huang I copied the document to google doc, and updated according to your suggestion. I think it's easier to see the whole picture in google doc.

Here is the link of google doc:

https://docs.google.com/document/d/1EoESyc-o9DNb8Es5R14BCAxtIafjWy1PiQCcI9VYoh8/edit?usp=sharing

**Comment by Jennifer88huang** [comment]
@codelipenghui Good idea. I've reviewed the Google doc, it's quite clear, and I have no issue with it.

**Comment by Jennifer88huang** [comment]
@codelipenghui Good idea. I've reviewed the Google doc, it's quite clear, and I have no issue with it.

**Comment by codelipenghui** [comment]
@Jennifer88huang Thanks, PR is updated.

**Comment by codelipenghui** [comment]
run Java8 Tests

**Comment by codelipenghui** [comment]
run Java8 Tests

**Pull request merged**